charts/cert-manager/test/test-certmanager.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cd $(dirname $0)/../../..
+source hack/lib.sh
+
+# create kind cluster
+export KIND_CLUSTER_NAME=cert-manager-tester
+export DISABLE_CLUSTER_EXPOSER=yes
+source hack/ci/setup-kind-cluster.sh
+
+# try to install cert-manager
+echodate "Installing cert-manager..."
+kubectl apply -f charts/cert-manager/crd/
+
+helm upgrade \
+  --install \
+  --namespace cert-manager \
+  --create-namespace \
+  --atomic \
+  --set cert-manager.clusterIssuers.letsencrypt-prod.email=dev@kubermatic.com \
+  --set cert-manager.clusterIssuers.letsencrypt-staging.email=dev@kubermatic.com \
+  cert-manager charts/cert-manager/
+
+echodate "Downloading cmctl..."
+OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -sLo cmctl.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/cmctl-$OS-$ARCH.tar.gz
+tar xzf cmctl.tar.gz
+
+function cmctl_cleanup {
+  echodate "Cleaning up..."
+  rm cmctl cmctl.tar.gz
+  exit $exitcode
+}
+trap cmctl_cleanup EXIT
+
+echodate "Testing cert-manager..."
+./cmctl check api --wait=2m
+exitcode=$?
+
+echodate "Deleting kind cluster..."
+kind delete cluster --name "$KIND_CLUSTER_NAME"
```

Comment:
One thing I noticed overall in the script: is there a specific reason why `set -euo pipefail` is not set? From my point of view I think it should be, because there is no real reason for the script to continue if it fails in the middle (e.g. at downloading cmctl).

If so of course we would need to move the kind cluster deletion into the cleanup, similar to how it is done here https://github.com/kubermatic/kubermatic/blob/master/hack/run-ccm-migration-e2e-test-in-kind.sh#L52

Reply:
Frankly, I just renamed it and changed the cmctl, didn't look close enough to nice it's not there :-) I added this now.

Reply:
Looks good so far. However the last line in the script which deletes the cluster now needs to be trapped as well. Because what can happen is that is that the script starts the cluster but fails in the middle. With the current code it would not run the cluster deletion. Additionally if I recall bash correctly you also need the `|| true` so the cleanup does not break the script, similar to here master/hack/run-ccm-migration-e2e-test-in-kind.sh#L52

Reply:
Yeah, I added the `|| true` and exit with correct code. I captured it, but didn't use it. I think that's why there was no `-eo pipefail` before and I removed it as well, because the only thing that we care about is handled anyway.

Reply:
Okay I think this is ok now. You are right, we are doing it in other places and it also fits our [bash guidelines](https://github.com/kubermatic/kubermatic/blob/weekly-2021-50/docs/code-style.md#guidelines). I am resolving the conversation now.